src/pages/index.vue

```diff
@@ -106,6 +112,19 @@ watch(
     selectDataForRoute
 );
 
+watch([countryToSelect, admin1Indicators], async () => {
+    // wait until admin1 loaded before selecting country
+    if (countryToSelect.value !== null && Object.keys(admin1Indicators.value).length) {
+        // set waiting flag so Choropleth can show spinner while Leaflet updates. Debounce the select country
+        // action so Vue can start showing spinner first.
+        waitingForMapBounds.value = true;
+        debounce(() => {
+            selectCountry(countryToSelect.value);
+            countryToSelect.value = null;
+        })();
```

Comment:
I don't follow what the issue is that we're routing around. Why doesn't the line `this.loading = true` (which is the first thing `selectCountry` does, unless we're un-selecting a country) take care of the spinner? -- it seems the update to `this.loading` would be executed before the change to selectedCountryId, which I take to be what triggers Leaflet updating?

```ts
    async selectCountry(countryId: string) {
            if (countryId === this.selectedCountryId) {
                return;
            }

            if (!countryId) {
                this.selectedCountryId = "";
                return;
            }

            this.loading = true;

            // Some countries do not have admin2 regions or data - if one of these is selected, we load
            // a more detailed geojson, and re-use its admin1 indicators as "admin2"
            const level = this.appConfig.countriesWithoutAdmin2.includes(countryId) ? 1 : 2;

            if (!(countryId in this.admin2Indicators)) {
                if (level === 1) {
                    this.admin2Indicators[countryId] = this.admin1Indicators[countryId];
                } else {
                    this.admin2Indicators[countryId] = await getIndicators(countryId, level);
                }
            }
            if (!(countryId in this.admin2Geojson)) {
                this.admin2Geojson[countryId] = await getGeojsonFeatures(countryId, level);
            }

            this.selectedCountryId = countryId;
            this.loading = false;
        }
```

Reply:
Oh, that bit works fine! The trouble is that we load everything and subsequently set loading to false, so the spinner goes away, but Vue+leaflet are still busy sorting their lives out for another (potentially) several seconds before the UI updates. So I wanted to prolong the time that the spinner was shown until view really had finished updating, and the only reliable place I found to hook into that was Leaflet decaring its bounds to have updated...

Reply:
This should become a moot point once Mantra works his optimisation magic... 🤞

Reply:
just reviewing this now, will probably be changing a little bit of this in the next PR! there is a debounce but elsewhere :laughing: